cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -230,6 +317,9 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
       BUILD Arrow
       VERSION ${VERSION}
       EXPORT_SET arrow_targets
+      # TODO: What about all the other GLOBAL_TARGETS specified in the `rapids_cpm_find` command?
+      # Also, should this be constructed conditionally based on whether we're using shared or static
+      # libs for arrow?
```

Comment:
As above, the conditional construction seems redundant, but mismatch with `rapids_cpm_find` above seems odd.

Reply:
The superset in the rapids_cpm_find is needed when you locate an existing built arrow. In that case all the optional arrow components like ArrowDataset are found in this single call so we need to mark all the targets as global.

Likewise when we are generating the components we mark the targets from that subcomponent as global to a have a clean mapping of components to targets.